src/main/java/com/openlattice/ids/IdsGeneratingEntryProcessor.java

```diff
@@ -44,7 +45,7 @@ public IdsGeneratingEntryProcessor( int count ) {
         final Range range = entry.getValue(); //Range should never be null in the EP.
         final UUID[] ids = getIds( range );
         entry.setValue( range );
-        return Arrays.asList( ids );
+        return DelegatedUUIDList.wrap( Arrays.asList( ids ) );
```

Comment:
Does it make any difference whether the function (/entry processor) returns a List<UUID> or DelegatedUUIDList?

Reply:
Yes, because just returning a list means it uses java serialization, but wrapping it users stream serializer instead.

Reply:
(I ask because the return type of the function is still List<UUID>)

Reply:
`List<UUID>`

Reply:
Hazelcast uses the class object not the type specified on the interface. Otherwise, it would have to use reflection to unreliably reify types on entry processors.